### PR TITLE
[FEATURE] Rule-Based Profiler: implement column name suffix filtering logic for column domain builder classes

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
1) Imbue all column-type domain builders with column name suffix processing (include/exclude) capabilities.
2) Imbue all column-type domain builders with semantic column type processing (include/exclude) capabilities.
3) Add "passthrough" semantic types for `CategoricalColumnDomainBuilder` which are allowed without computation.
4) Add "exclusion" semantic types for `CategoricalColumnDomainBuilder` based on reasonable cardinality assumptions.
5) Delete obsolete `simple_suffix_column_domain_builder.py`.
6) Update unit tests.
7) Clean up / rearrange for better readability order of arguments in `DomainBuilder` class initializers.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-464/GREAT-682
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Review by NathanFarmer — APPROVED** [review]
LGTM!

**Comment by NathanFarmer** [comment]
This will give us much more flexibility!

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py [hunk 1]
```diff
@@ -202,41 +242,77 @@ def get_effective_column_names(
...
+            else:
+                if not isinstance(exclude_column_name_suffixes, (Iterable, list)):
+                    raise ValueError(
+                        "Unrecognized exclude_column_name_suffixes directive -- must be a list or a string."
+                    )
+
+            effective_column_names: List[str] = list(
+                filter(
+                    lambda candidate_column_name: not candidate_column_name.endswith(
+                        tuple(include_column_name_suffixes)
```
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py]: Should this instead be 
```suggestion
                        tuple(exclude_column_name_suffixes)
```

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
LGTM except for what looks like a typo

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py]: Fixed -- and committed -- and added a test -- thanks so much, @anthonyburdi !

**Review by anthonyburdi — APPROVED** [review]
LGTM!

**Pull request merged**